luxtronik/__main__.py

```diff
@@ -0,0 +1,37 @@
+"""Luxtronik CLI."""
+import argparse
+import sys
+from luxtronik.discover import discover as _discover
+from luxtronik.scripts.dump import dump  # pylint: disable=unused-import # noqa: F401
+
+
+def discover():
+    """Nicely printing discovery results."""
+    results = _discover()
+    print(f"{len(results)} heatpump(s) reported back")
+    for result in results:
+        print(f"IP Address: {result[0]} Port: {result[1]}")
+
+
+def main() -> int:
+    """Command parser."""
+    parser = argparse.ArgumentParser(
+        description="CLI for Luxtronik controllers",
+        usage="""luxtronik <command> [<args>]
+        The supported commands are:
+        dump       Dump all data from the luxtronik controller 
+        discover   Send discovery message and print reults 
+        """,
+    )
+    parser.add_argument("command", help="Subcommand to run")
+    # parse_args defaults to [1:] for args, but you need to
+    # exclude the rest of the args too, or validation will fail
+    args = parser.parse_args(sys.argv[1:2])
+    if args.command not in ("dump", "discover"):
+        print("Unrecognized command")
+        parser.print_help()
+        sys.exit(1)
+    # pop command, otherwise the argparser within the called script will fail
+    sys.argv.pop(1)
+    # invoke method with same name as the command, dump -> dump() for example
+    globals()[args.command]()
```

Comment:
That seems a little bit hacky. Maybe we should maintain a map that could be used rather than using globals()?

Reply:
🤔 you got a point there ... I'll push a change, maybe over the weekend if I fin some time for that